### PR TITLE
Docker Feature

### PR DESCRIPTION
**Pull request by Killerherts**
Pulled old PR because of bad history and messy commit history.

Pushed new image with changes to killerherts/bing-rewards:test  

Note. container will hang when running BingRewards.py this is because of my changes in driver.py not being part of the master.
you will also have to include full path to setup.py if running from docker run as setup is not moved to BingRewards folder  in the master branch as well

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,15 @@ actually translates to `python BingRewards.py -r -hl -d chrome`, i.e run the rem
 Here's an example of running non-default arguments
 `python BingRewards.py -w -nhl -e my_email@gmail.com -p`, i.e run web searches in non-headless mode with specified email, the password will be prompted for separately.
 
+## Container Installation
+1. In terminal, run `docker pull killerherts/bing-rewards:latest`
+2. Set-up the config with either option 1 or 2 
+	1. Option 1, run setup.py again: `docker run -t -d --name bing-rewards killerherts/bing-rewards:latest python setup.py -e <your_email> -p <password>`  You must include your password as there will be no user prompt with -t -d
+	2. Option 2: Pass your config volume directly into the container: `docker run -t -d -v <absolute-path-to-config-directory>:/config --name bing-rewards killerherts/bing-rewards:latest`
+4. To enter the container for maintance docker exec -it bing-rewards /bin/bash
```
Comment (pironic) [inline README.md:44]: hm looks like the markdown got lost from the last pull...? 
`docker exec -it bing-rewards /bin/bash` to ``` `docker exec -it bing-rewards /bin/bash` ```

Comment (Killerherts) [reply README.md]: Corrected

**Review by pironic — APPROVED** [review]

**Comment by CptanPanic** [comment]
Any updates on this ?

**Comment by Killerherts** [comment]
> Any updates on this ?
Just waiting for review and merge.

**Comment by jjjchens235** [comment]
Thanks for the updates, this looks good, I will make a few final changes to the README and then merge to master.

I see one minor issue that might need to be fixed in the future. There is no way to update chromium versions. Ideally this would be auto-updated in the cron script, but it cannot be even done manually because `sudo` isn't part of the image.

**Comment by Killerherts** [comment]
> Thanks for the updates, this looks good, I will make a few final changes to the README and then merge to dev/master. FYI, the main change to the README is that it instructs the user to pull from `:test` rather than `:latest`. Once you update `:latest` then can you update the README as well?
> 
> I see one minor issue that might need to be fixed in the future. There is no way to update chromium versions. Ideally this would be auto-updated in the cron script, but it cannot be even done manually because `sudo` isn't part of the image.
> 
> If the user pulls the image but does not download a driver immediately, it could effectively lead to a correct driver never being able to be downloaded. For example, if I pulled the image **1 month ago** where chromium is at v103. Let's say I actually do a `docker run` **today**, `Driver.py` tries to download last 3 versions of driver which is 106/105/104, it will never download v103, and the program will never be able to be run.
> 
> Recap:
> 
> 1. I will merge to dev remote
> 2. Please review updated README commit that I made to make sure it makes sense
> 3. I will merge to master if it looks good to you
> 4. At your convenience, please update `:latest` in dockerhub and  the `docker pull` in README
> 5. At your convenience, within the image, please provide a way to update chromium manually at the least,  and optionally automatically as well

That's a really good point just adding in sudo is probably the path of least resistance unless someone has a more elegant solution.

**Comment by jjjchens235** [comment]
We good to go on the `dev` branch?

**Comment by Killerherts** [comment]
> I am trying this out, and encountered this minor issue. Following the readme, I mapped directory to /bing to get logs, but it doesn't work because '+ ln -s /bing-rewards/BingRewards/logs /bing' does not work, because if that directory already exists as it would if you map it, then it just puts a link inside of it to logs which doesn't end up writing logs to filesystem. Probably want to do the opposite like 'ln -s /bing /bing-rewards/BingRewards/logs' Otherwise I was able to also get it to work by mapping to /bing-rewards/BingRewards/logs
> 
> Regarding the embedded cron, which is a good idea for long term running, but it would be nice if it ran by default when you start container also, so would adding '@reboot /bin/bash /bing-rewards/script.sh > $logfile 2>&1' work in the crontab?

I had not updated latest image as I had been away from my desk. I just pushed latest a few minutes ago I will also make my dev tag pull the dev tag the same way.

**Comment by Killerherts** [comment]
I have figured out GitHub actions and set it up to auto rebuild every day. We no longer have to worry about underlying dependencies not getting updated. No code needs to be changed, as this is just scripting on my repo.

**Comment by jjjchens235** [comment]
Why does it need to follow either of our repo's ? I thought the container logic is to do a clone?

Are we good for me to push to master, or are there some open items? No rush, but want to make sure I'm not a bottleneck here.

**Comment by Killerherts** [comment]
> Why does it need to follow either of our repo's ? I thought the container logic is to do a clone?
> 
> Are we good for me to push to master, or are there some open items? No rush, but want to make sure I'm not a bottleneck here.

It's only follows my repo. I just have to code the function to check if your repo has been pushed this is all on my repo won't effect yours. I just ran out of time to work on the check function on your repo. 

This is good to push to master I have been running it and pulling dev for over week without any issues. 

Thanks

**Comment by jjjchens235** [comment]
> > Why does it need to follow either of our repo's ? I thought the container logic is to do a clone?
> > Are we good for me to push to master, or are there some open items? No rush, but want to make sure I'm not a bottleneck here.
> 
> It's only follows my repo. I just have to code the function to check if your repo has been pushed this is all on my repo won't effect yours. I just ran out of time to work on the check function on your repo.
> 
> This is good to push to master I have been running it and pulling dev for over week without any issues.
> 
> Thanks

Which tag should the README state? currently readme states `docker pull killerherts/bing-rewards:test` in dev.

**Comment by Killerherts** [comment]
The :dev tag pulls your dev branch. The :latest tag pulls the master branch.

**Comment by jjjchens235** [comment]
@Killerherts  do you have a chance to look at this issue involving issue with docker ?

https://github.com/jjjchens235/bing-rewards/issues/290

**Comment by Killerherts** [comment]
@jjjchens235 I am setting up an environment var to solve the TZ. I am of the opinion we should rewrite this to not clone your repo during the build process as it this is creating the issue they are having with placing logs.

The reason we did this was to prevent the container for falling out of sync with updates, I can now solve that problem with GitHub actions and a bit of script. It will take longer to make the changes but I believe it will provide a better user experience and make changing in the future far easier to understand if someone ends up taking my place. I clone the repo during the build process essentially erases any local changes you might have made.  

If you agree with this, I can work it up this week I am partway through implementing this but don't want to progress too far if you're not in agreement with this solution.

**Comment by jjjchens235** [comment]
> @jjjchens235 I am setting up an environment var to solve the TZ. I am of the opinion we should rewrite this to not clone your repo during the build process as it this is creating the issue they are having with placing logs.
> 
> The reason we did this was to prevent the container for falling out of sync with updates, I can now solve that problem with GitHub actions and a bit of script. It will take longer to make the changes but I believe it will provide a better user experience and make changing in the future far easier to understand if someone ends up taking my place. I clone the repo during the build process essentially erases any local changes you might have made.
> 
> If you agree with this, I can work it up this week I am partway through implementing this but don't want to progress too far if you're not in agreement with this solution.

Makes sense to me!

**Pull request merged**